client/onboarding/add-business-info-task/index.js

```diff
@@ -0,0 +1,171 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useState, useContext } from 'react';
+import { __ } from '@wordpress/i18n';
+import { SelectControl, Card, CardBody } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import WizardTaskItem from 'wcpay/additional-methods-setup/wizard/task-item';
+import WizardTaskContext from 'wcpay/additional-methods-setup/wizard/task/context';
+import OnboardingSelectControl from 'wcpay/components/onboarding-select-control';
+import { useBusinessTypes } from 'wcpay/data/onboarding';
+import strings from '../strings';
+
+const AddBusinessInfo = () => {
+	const { setCompleted } = useContext( WizardTaskContext );
+	const { businessTypes, isLoading } = useBusinessTypes();
+	const {
+		connect: { availableCountries, country },
+	} = wcpaySettings;
+
+	const countries = Object.keys( availableCountries ).map(
+		( countryCode ) => {
+			return {
+				label: availableCountries[ countryCode ],
+				value: countryCode,
+			};
+		}
+	);
+
+	const [ businessCountry, setBusinessCountry ] = useState( country );
+	const [ businessType, setBusinessType ] = useState();
+	const [ businessStructure, setBusinessStructure ] = useState();
+	const [ displayStructures, setDisplayStructures ] = useState( false );
+
+	const getBusinessTypesForCountry = ( countryCode ) => {
+		if ( isLoading || ! businessTypes.hasOwnProperty( 'data' ) ) {
+			return false;
+		}
+
+		return businessTypes.data.hasOwnProperty( countryCode )
+			? businessTypes.data[ countryCode ]
+			: false;
+	};
+
+	const formatBusinessTypes = ( countryCode ) => {
+		const businessTypeData = getBusinessTypesForCountry( countryCode );
+
+		if ( isLoading || ! businessTypeData ) {
+			return [];
+		}
+
+		return Object.keys( businessTypeData ).map( ( key ) => ( {
+			name: strings.businessTypes[ key ],
+			description: 'Test',
+			key: key,
+		} ) );
+	};
```

Comment:
Following up with my previous comment. We can then move this function and all that are similar outside the component.

With functional components we should think that all the component will be called again, meaning that all of these function will be declared every time.

That's when `useCallback` and `useMemo` can be useful. But if we can just move them outside, will be better 🙂